### PR TITLE
#161779823 implement reset password feature

### PR DESCRIPTION
**Pull request by idmega2000**
#### What does this PR do?
It implements the reset password feature
#### Description of Task to be completed?
- [x] send user reset password link
- [x] update user password
- [x] refactor form container and form wrapper
#### How should this be manually tested?
- npm run start and visit ``` http://localhost:8080/reset-password```
#### What are the relevant pivotal tracker stories?
[161779823](https://www.pivotaltracker.com/story/show/161779823)

#### Questions:
None

### CODE REVIEW

File: src/containers/Auth/ResetPassword/UpdatePassword.js [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+
+  /**
+   * @function
+   * @returns {JSX} The update password form JSX
+   * or the update password successful JSX
+   */
+  render() {
+    return (
+        <FormWrapper imageUrl={image}
+        imgId='updateSideImage' displayImage={false}>
```
Comment (JohnMadakin) [inline src/containers/Auth/ResetPassword/UpdatePassword.js]: can you change the prop name to `imageId` or something more readable

File: src/containers/Auth/ResetPassword/__test__/updatePassword.spec.js [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+  });
+
+
+  it('should render the FormContainer', () => {
+    const container = wrapper.find('FormContainer');
+    expect(container.length).toEqual(1);
+  });
+
+
+  it('should redirect if got to signin button is clicked', () => {
```
Comment (ekpangmichael) [inline src/containers/Auth/ResetPassword/__test__/updatePassword.spec.js]: Awesome job so far @idmega2000, just fix this minor typo  here ``` got ```

File: src/containers/Auth/ResetPassword/UpdatePassword.js (hunk 1 above)
Comment (idmega2000) [reply src/containers/Auth/ResetPassword/UpdatePassword.js]: Thank you for the feedback. imgId was added as a prop on the Formwrapper component. Changing it now would affect all the files that use the component. We would take note of this during refactoring.

File: src/containers/Auth/ResetPassword/ResetPassword.js [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+   */
+  handleSubmit = (e) => {
+    e.preventDefault();
+    const { sendResetLink } = this.props;
+    sendResetLink(this.state.email);
+  }
+
+  /**
+   * @function
+   * @returns {JSX} The reset password form JSX
```
Comment (Danielshow) [inline src/containers/Auth/ResetPassword/ResetPassword.js]: @idmega2000 The return `description` is not descriptive

File: src/containers/Auth/ResetPassword/ResetPassword.js [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+                </div>
+                <div className='text-center' id='clipLoader'>
+                {!this.props.isLoading
+                  && <Button
+                  type={'submit'} buttonId='submitButton'
+                  classes='form-button' >
+                  Send Reset Password Link</Button>}
+
+                  <ClipLoader
+                    css={'override'}
```
Comment (Danielshow) [inline src/containers/Auth/ResetPassword/ResetPassword.js]: @idmega2000 is there an `override` class? if not, you can remove the css field.
`size unit` can just be `px` without curly brace

File: src/containers/Auth/ResetPassword/UpdatePassword.js (hunk 1 above)
Comment (Danielshow) [reply src/containers/Auth/ResetPassword/UpdatePassword.js]: @idmega2000, you can change that, we only have two features merged

File: src/containers/Auth/ResetPassword/UpdatePassword.js [hunk 5]
```diff
@@ -0,0 +1,212 @@
...
+             {!this.props.isLoading && <Button type={'submit'}
+             buttonId='updateButton'
+                classes='form-button' >
+                  Change Password</Button>}
+                  <p className='password-inst'>
+              *Password must contain at least a number and must
+                not be less than 8 characters</p>
+                  <ClipLoader
+                    css={'override'}
+                    sizeUnit={'px'}
```
Comment (Danielshow) [inline src/containers/Auth/ResetPassword/UpdatePassword.js]: same here `override` and `curly braces`

Comment (idmega2000) [reply src/containers/Auth/ResetPassword/UpdatePassword.js]: Thanks. it has been fixed

File: src/containers/Auth/ResetPassword/ResetPassword.js [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+  handleInputChange = (e) => {
+    this.setState({
+      [e.target.id]: e.target.value,
+    });
+  }
+
+  /**
+   * @description - decides if component should throw error or update
+   * @param {*} nextProps
+   * @param {*} nextState
```
[the remark this answers is not in the document]
Comment (idmega2000) [reply src/containers/Auth/ResetPassword/ResetPassword.js]: Thanks. I have fix this

**Pull request merged**